src/Microsoft.AspNet.Mvc.ViewFeatures/Rendering/Html/DefaultEditorTemplates.cs

```diff
@@ -378,7 +378,11 @@ public static string TimeInputTemplate(IHtmlHelper htmlHelper)
 
         public static string NumberInputTemplate(IHtmlHelper htmlHelper)
         {
-            return GenerateTextBox(htmlHelper, inputType: "number");
+            var htmlAttributes =
+                CreateHtmlAttributes(htmlHelper, className: "text-box single-line", inputType: "number");
+            htmlAttributes["step"] = "any";
```

Comment:
You are right. I was investigating #2780 and it is possible that we might not even need this change based on what we decide for that scenario.